src/cascadia/QueryExtension/ExtensionPalette.cpp

```diff
@@ -0,0 +1,574 @@
+        AIContentDisclaimerHyperlink().Inlines().Append(disclaimerLinkText);
+
+        auto learnMoreLinkText = Windows::UI::Xaml::Documents::Run();
+        learnMoreLinkText.Text(RS_(L"LearnMoreLink"));
+        LearnMoreLink().Inlines().Append(learnMoreLinkText);
+
+        _loadedRevoker = Loaded(winrt::auto_revoke, [this](auto /*s*/, auto /*e*/) {
+            // We have to add this in (on top of the visibility change handler below) because
+            // the first time the palette is invoked, we get a loaded event not a visibility event.
+
+            // Only let this succeed once.
+            _loadedRevoker.revoke();
+
+            _setFocusAndPlaceholderTextHelper();
+
+            // For the purposes of data collection, request the API key/endpoint *now*
+            _AIKeyAndEndpointRequestedHandlers(nullptr, nullptr);
+
+            TraceLoggingWrite(
+                g_hQueryExtensionProvider,
+                "QueryPaletteOpened",
+                TraceLoggingDescription("Event emitted when the AI chat is opened"),
+                TraceLoggingBoolean((!_AIKey.empty() && !_AIEndpoint.empty()), "AIKeyAndEndpointStored", "True if there is an AI key and an endpoint stored"),
+                TraceLoggingKeyword(MICROSOFT_KEYWORD_CRITICAL_DATA),
+                TelemetryPrivacyDataTag(PDT_ProductAndServiceUsage));
+        });
+
+        // Whatever is hosting us will enable us by setting our visibility to
+        // "Visible". When that happens, set focus to our query box.
+        RegisterPropertyChangedCallback(UIElement::VisibilityProperty(), [this](auto&&, auto&&) {
+            if (Visibility() == Visibility::Visible)
+            {
+                // Force immediate binding update so we can select an item
+                Bindings->Update();
+
+                _setFocusAndPlaceholderTextHelper();
+
+                // For the purposes of data collection, request the API key/endpoint *now*
+                _AIKeyAndEndpointRequestedHandlers(nullptr, nullptr);
+
+                TraceLoggingWrite(
+                    g_hQueryExtensionProvider,
+                    "QueryPaletteOpened",
+                    TraceLoggingDescription("Event emitted when the AI chat is opened"),
+                    TraceLoggingBoolean((!_AIKey.empty() && !_AIEndpoint.empty()), "AIKeyAndEndpointStored", "Is there an AI key and an endpoint stored"),
+                    TraceLoggingKeyword(MICROSOFT_KEYWORD_CRITICAL_DATA),
+                    TelemetryPrivacyDataTag(PDT_ProductAndServiceUsage));
+            }
+            else
+            {
+                _close();
+            }
+        });
+
+        // We do not have a color ramp to support this.
+        RequestedTheme(ElementTheme::Dark);
+    }
+
+    void ExtensionPalette::AIKeyAndEndpoint(const winrt::hstring& endpoint, const winrt::hstring& key)
+    {
+        _AIEndpoint = endpoint;
+        _AIKey = key;
+        _httpClient = winrt::Windows::Web::Http::HttpClient{};
```

Comment:
probably don't need to recreate this every time these change, but I'm OK with it for now